### ISSUE TITLE
`index` is incorrectly displayed in case printing DataFrame with `omnisci` storage format

### ISSUE THREAD

**Issue by prutskov** [modin-project/modin#4143]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 
- **Modin version** (`modin.__version__`): 0.13
- **Python version**: 3.9.10
- **Code we can use to reproduce**:

```python
import modin.pandas as pd
import modin.config as cfg
cfg.StorageFormat.put("omnisci")

columns = [
        "vendor_id",
        "pickup_datetime",
        "dropoff_datetime",
        "passenger_count",
        "trip_distance",
        "pickup_longitude",
        "pickup_latitude",
        "rate_code",
        "store_and_fwd_flag",
        "dropoff_longitude",
        "dropoff_latitude",
        "payment_type",
        "fare_amount",
        "surcharge",
        "mta_tax",
        "tip_amount",
        "tolls_amount",
        "total_amount"]

df = pd.read_csv("nyc_taxi_data.csv", parse_dates=["pickup_datetime", "dropoff_datetime"], names=columns, header=0)
print(df)
print(df.index)
```

Output:
```bash
   vendor_id     pickup_datetime  ... tolls_amount  total_amount
0        CMT 2014-01-09 20:45:25  ...         0.00          8.90
1        CMT 2014-01-09 20:46:12  ...         0.00         11.40
2        CMT 2014-01-09 20:44:47  ...         0.00         14.00
3        CMT 2014-01-09 20:44:57  ...         0.00         10.20
4        CMT 2014-01-09 20:47:09  ...         0.00          8.75
..       ...                 ...  ...          ...           ...
57       CMT 2014-12-26 00:54:43  ...         0.00          7.00
58       CMT 2014-12-26 19:35:43  ...         5.33         41.79
59       CMT 2014-12-25 23:27:36  ...         0.00         63.00
60       CMT 2014-12-25 23:52:17  ...         0.00         17.40
61       CMT 2014-12-26 03:20:07  ...         0.00         57.60

[165114361 rows x 18 columns]

RangeIndex(start=0, stop=165114361, step=1)
```

Data is NY Taxi(https://s3-us-west-2.amazonaws.com/nyctlc/nyc_taxi_data.csv.gz). Decompressed csv is used.

### Describe the problem

`print(df)` provides incorrect view of `df.index`, but property `index` has correct values.

**Comment by vnlitvinov** [comment]
@modin-project/modin-omnisci  could someone please check if this still happens?

**Comment by AndreyPavlenko** [comment]
The issue seems very similar to #4107, which is already fixed.
On the latest master the above code gives me the following output (Note: I've truncated the data file to 100000 lines):

      vendor_id     pickup_datetime  ... tolls_amount  total_amount
0           CMT 2014-01-09 20:45:25  ...          0.0          8.90
1           CMT 2014-01-09 20:46:12  ...          0.0         11.40
2           CMT 2014-01-09 20:44:47  ...          0.0         14.00
3           CMT 2014-01-09 20:44:57  ...          0.0         10.20
4           CMT 2014-01-09 20:47:09  ...          0.0          8.75
...         ...                 ...  ...          ...           ...
99995       CMT 2014-01-10 13:31:26  ...          0.0          5.85
99996       CMT 2014-01-10 13:26:26  ...          0.0         17.00
99997       CMT 2014-01-10 13:21:11  ...          0.0         33.00
99998       CMT 2014-01-10 13:27:16  ...          0.0          5.00
99999       CMT 2014-01-10 07:55:08  ...          0.0          8.40

[100000 rows x 18 columns]
RangeIndex(start=0, stop=100000, step=1)

**Comment by AndreyPavlenko** [comment]
Closing this issue as resolved by #4834. If you still can reproduce the issue, please reopen.
